src/sema/tests/mod.rs

```diff
@@ -128,3 +139,193 @@ fn test_statement_reachable() {
         assert_eq!(test_case.reachable(), expected);
     }
 }
+
+#[test]
+fn constant_overflow_checks() {
+    let file = r#"
+    contract test_contract {
+        function test_params(uint8 usesa, int8 sesa) public returns(uint8) {
+            return usesa;
+        }
+    
+        function test_add(int8 input) public returns (uint8) {
+            // value 133 does not fit into type int8.
+            int8 add_ovf = 127 + 6;
+    
+            // negative value -1 does not fit into type uint8. Cannot implicitly convert signed literal to unsigned type.
+            uint8 negative = 3 - 4;
+    
+            // value 133 does not fit into type int8.
+            int8 mixed = 126 + 7 + input;
+    
+            // negative value -1 does not fit into type uint8. Cannot implicitly convert signed literal to unsigned type.
+            return 1 - 2;
+        }
+    
+        function test_mul(int8 input) public {
+            // value 726 does not fit into type int8.
+            int8 mul_ovf = 127 * 6;
+    
+            // value 882 does not fit into type int8.
+            int8 mixed = 126 * 7 * input;
+        }
+    
+        function test_shift(int8 input) public {
+            // value 128 does not fit into type int8.
+            int8 mul_ovf = 1 << 7;
+    
+            // value 128 does not fit into type int8.
+            int8 mixed = (1 << 7) + input;
```

Comment:
There are no tests that include shift right and divide.
For example,
`uint8 b = 255 - 255/5` should not raise any error.